pkg/config/validation/validation.go

```diff
@@ -2667,6 +2667,55 @@ func analyzeUnreachableHTTPRules(routes []*networking.HTTPRoute,
 			}
 		}
 	}
+
+	validateByPassedRouteForCatchAll(routes, reportUnreachable)
+}
+
+func validateByPassedRouteForCatchAll(routes []*networking.HTTPRoute, reportUnreachable func(ruleno string, reason string)) {
+	catchall := false
+	if len(routes) == 1 && (len(routes[0].Match) == 1 || len(routes[0].Match) == 0) {
+		return
+	}
+	for rulen, route := range routes {
+		if len(route.Match) == 0 && route.Mirror == nil && route.Fault == nil {
+			catchall = true
+		} else {
+			for i, match := range route.Match {
+				if isCatchAllMatch(match) && i != len(route.Match)-1 {
+					catchall = true
+					break
+				}
+			}
+		}
+		if catchall {
+			reportUnreachable(routeName(route, rulen), "unreachable route detected, as a catch-all or default route is defined above")
+		}
+	}
+}
+
+func isCatchAllMatch(m *networking.HTTPMatchRequest) bool {
```

Comment:
can we resue that from pilot/pkg/networking/core/v1alpha3/route/route.go

Reply:
Oh, that one can be moved here as not used there

Reply:
`isCatchAllMatch` is actually copied from ` pilot/pkg/networking/core/v1alpha3/route/route.go`, it's being used for test cases only and can't import it due to circular dependency. Maybe to put this method in some util so it can be shared by both, any suggestion where to put?

Reply:
test case can be moved too

Reply:
Removed `isCatchAllMatch` in `route.go` but still keep those test cases